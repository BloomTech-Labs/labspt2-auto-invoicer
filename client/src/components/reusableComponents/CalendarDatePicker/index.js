import React, { Component } from "react";
import "./CalendarDatePicker.css";

import "react-dates/initialize";
import "react-dates/lib/css/_datepicker.css";

import { SingleDatePicker } from "react-dates";

class CalendarDatePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      date: null,
      focused: null
    };
  }

  render() {
    return (
      <div>
        <SingleDatePicker
          date={this.state.date}
          onDateChange={date => this.setState({ date })}
          focused={this.state.focused}
          onFocusChange={({ focused }) => this.setState({ focused })}
        />
      </div>
    );
  }
}

export default CalendarDatePicker;
