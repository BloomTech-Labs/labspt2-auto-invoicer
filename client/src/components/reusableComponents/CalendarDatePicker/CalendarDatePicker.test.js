import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import CalendarDatePicker from "../CalendarDatePicker";

describe("<CalendarDatePicker />", () => {
  it("renders Calendar Date Picker", () => {
    const { asFragment } = render(<CalendarDatePicker />);
    expect(asFragment()).toMatchSnapshot();
  });
});
