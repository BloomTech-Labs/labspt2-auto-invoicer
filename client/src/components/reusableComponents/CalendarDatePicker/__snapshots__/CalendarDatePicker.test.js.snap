// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarDatePicker /> renders Calendar Date Picker 1`] = `
<DocumentFragment>
  <div>
    <div
      class="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          class="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <div
            class="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-date"
              aria-label="Date"
              autocomplete="off"
              class="DateInput_input DateInput_input_1"
              id="date"
              name="date"
              placeholder="Date"
              type="text"
              value=""
            />
            <p
              class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-date"
            >
              Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
